### PR TITLE
BCStateTran: Add a periodic status log while in GettingMissingBlocks

### PR DESCRIPTION
**Pull request by glevkovich**
State Transfer Improvements:
1) util: Introduce a new class Throughput - we will first use it in BCStateTran, but it is for general use.
2) BCStateTran: Remove unneeded metric
3) BCStateTran: Add a periodic status log while in GettingMissingBlocks.
4) BCStateTran: Add throughput, blocks, and bytes to metrics report

### CODE REVIEW

File: tests/apollo/test_skvbc_ro_replica.py [hunk 1]
```diff
@@ -163,7 +163,7 @@ async def test_ro_replica_start_with_delay(self, bft_network, tracker):
         # await tracker.skvbc.tracked_fill_and_wait_for_checkpoint(
         # initial_nodes=bft_network.all_replicas(),
         # num_of_checkpoints_to_add=1)
-        with trio.fail_after(seconds=60):
+        with trio.fail_after(seconds=100):
```
Comment (toly-kournik) [inline tests/apollo/test_skvbc_ro_replica.py:166]: Why have you changed this test?

File: bftengine/src/bcstatetransfer/BCStateTran.hpp [hunk 2]
```diff
@@ -403,11 +406,34 @@ class BCStateTran : public IStateTransfer {
...
   concord::util::CallbackRegistry<uint64_t> on_transferring_complete_cb_registry_;
+
+  ///////////////////////////////////////////////////////////////////////////
+  // Internal Statistics
+  ///////////////////////////////////////////////////////////////////////////
+ protected:
+  static constexpr uint32_t get_missing_blocks_summary_window_size = checkpointWindowSize;
+  Throughput blocks_collected_;
+  Throughput bytes_collected_;
+  std::optional<uint64_t> first_collected_block_num_;
```
Comment (toly-kournik) [inline bftengine/src/bcstatetransfer/BCStateTran.hpp:431]: why optional? I think we count blocks from 1

File: tests/apollo/test_skvbc_ro_replica.py (hunk 1 above)
Comment (glevkovich) [reply tests/apollo/test_skvbc_ro_replica.py:166]: didn't mean to. Will be reverted.

File: bftengine/src/bcstatetransfer/BCStateTran.cpp [hunk 3]
```diff
@@ -1931,6 +1960,68 @@ void BCStateTran::EnterGettingCheckpointSummariesState() {
   sendAskForCheckpointSummariesMsg();
 }
 
+void BCStateTran::reportGettingMissingBlocksSummary(const uint64_t firstRequiredBlock, const uint32_t actualBlockSize) {
```
Comment (toly-kournik) [inline bftengine/src/bcstatetransfer/BCStateTran.cpp:1963]: A misleading name, isn't it?

File: bftengine/src/bcstatetransfer/BCStateTran.cpp [hunk 4]
```diff
@@ -1931,6 +1960,68 @@ void BCStateTran::EnterGettingCheckpointSummariesState() {
...
+    metrics_.prev_win_blocks_collected_.Get().Set(prev_win_block_results.num_processed_items_);
+    metrics_.prev_win_blocks_throughtput_.Get().Set(prev_win_block_results.throughput_);
+    metrics_.prev_win_bytes_collected_.Get().Set(prev_win_bytes_results.num_processed_items_);
+    metrics_.prev_win_bytes_throughtput_.Get().Set(prev_win_bytes_results.throughput_);
+
+    logGettingMissingBlocksSummary(firstRequiredBlock);
+  }
+}
+
+void BCStateTran::logGettingMissingBlocksSummary(const uint64_t firstRequiredBlock) {
```
Comment (toly-kournik) [inline bftengine/src/bcstatetransfer/BCStateTran.cpp:1986]: Same

File: bftengine/src/bcstatetransfer/BCStateTran.cpp (hunk 3 above)
Comment (glevkovich) [reply bftengine/src/bcstatetransfer/BCStateTran.cpp:1963]: changed to `reportCollectingStatus`

File: bftengine/src/bcstatetransfer/BCStateTran.cpp (hunk 4 above)
Comment (glevkovich) [reply bftengine/src/bcstatetransfer/BCStateTran.cpp:1986]: changed to `logColletingStatus`

**Comment by glevkovich** [comment]
@toly-kournik I've made all changes required changes.

File: bftengine/src/bcstatetransfer/BCStateTran.hpp (hunk 2 above)
Comment (toly-kournik) [reply bftengine/src/bcstatetransfer/BCStateTran.hpp:431]: Can you please elaborate on this?

**Review by toly-kournik — APPROVED** [review]
LGTM @glevkovich, Thanks!

**Pull request merged**